test/unittest/test_lib_utils.py

```diff
@@ -0,0 +1,36 @@
+import unittest
+
+import socket
+
+from hypothesis import given, settings
+from hypothesis.strategies import integers
+
+import lib.utils as utils
+
+
+class TestUtils(unittest.TestCase):
+    def test_extract_schema_from_snapshot(self):
+        snapshot_path = '00000000000000000003.snap'
+        v = utils.extract_schema_from_snapshot(snapshot_path)
+        # assertListEqual() is available since 3.1
+        is_equal = v == [u'version', 2, 3, 1]
+        self.assertTrue(is_equal)
+
+    @settings(max_examples=5)
+    @given(port=integers(65000, 65535))
+    def test_check_port(self, port):
+        def open_socket(p):
+            s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+            s.bind(('localhost', p))
+            s.listen(0)
+            return s
+        status = utils.check_port(port, False, ipv4=True, ipv6=False)
+        self.assertEqual(status, True)
+        s = open_socket(port)
+        status = utils.check_port(port, False, ipv4=True, ipv6=False)
+        s.close()
+        self.assertEqual(status, False)
```

Comment:
(It is my pet peeve, so I can't pass away. It is not a review comment and not something against your change.)

To be honest, I would prefer to [put it to the torch](https://github.com/tarantool/test-run/issues/141). Not sure, thought, it'll be possible entirely: at least there is 1.10 w/o logging of an actual port and we likely will meet other problem on the way.

The test depends on the environment: if someone listens on such port, the test will fail. However the range that you choosen should more or less protect us from this.

It also will fail with some probability if many instances of the test will be run in parallel or even one-by-one (AFAIR, `TIME_WAIT` is around two minutes and we don't use `SO_REUSEADDR` / zero `SO_LINGER` here). Hopefully, we'll not run the test several times during a short time. However we should remember: the test is flaky by design. If it'll be problematic, we unlikely will be able to propose a full fix.

Anyway, I don't mind adding the test case. Just... it is my old fault: it would be better if I would drop this machinery in 2017, when I worked on test-run parallelization.

BTW, you can safely drop the `rais` parameter: it is not used anywhere.

Reply:
> BTW, you can safely drop the rais parameter: it is not used anywhere.

Didn't get it.

Reply:
The second argument of the `check_port()` function is not used anywhere (neither in test-run, nor in tarantool's python tests).

Reply:
removed

Reply:
I meant to remove it from the source code, however, okay: it is out of scope here.